enterprise/frontend/src/embedding-sdk/components/public/NewQuestion/NewQuestion.tsx

```diff
@@ -0,0 +1,108 @@
+import { useState } from "react";
+import { usePreviousDistinct } from "react-use";
+
+import { withPublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
+import {
+  Notebook,
+  QuestionVisualization,
+} from "embedding-sdk/components/public/InteractiveQuestion/components";
+import {
+  InteractiveQuestionProvider,
+  useInteractiveQuestionContext,
+} from "embedding-sdk/components/public/InteractiveQuestion/context";
+import {
+  SaveQuestionForm,
+  SaveQuestionTitle,
+} from "metabase/components/SaveQuestionForm";
+import { SaveQuestionProvider } from "metabase/components/SaveQuestionForm/context";
+import { useCreateQuestion } from "metabase/query_builder/containers/use-create-question";
+import { useSaveQuestion } from "metabase/query_builder/containers/use-save-question";
+import { Button, Group, Stack, Tabs, Title, Box } from "metabase/ui";
+
+const SaveQuestion = ({ onClose }: { onClose: () => void }) => {
+  const { question, originalQuestion } = useInteractiveQuestionContext();
+
+  const handleCreate = useCreateQuestion();
+  const handleSave = useSaveQuestion();
+
+  if (!question) {
+    return null;
+  }
+
+  return (
+    <SaveQuestionProvider
+      question={question}
+      originalQuestion={originalQuestion}
+      onCreate={handleCreate}
+      onSave={handleSave}
+      multiStep={false}
+      initialCollectionId={null}
+    >
+      <Stack p="md">
+        <Title>
+          <SaveQuestionTitle />
+        </Title>
+        <SaveQuestionForm onCancel={onClose} />
+      </Stack>
+    </SaveQuestionProvider>
+  );
+};
+
+const NewQuestionInner = () => {
+  const [activeTab, setActiveTab] = useState<
+    "notebook" | "visualization" | "save" | null | (string & unknown)
+  >("notebook");
+
+  const previousTab = usePreviousDistinct(activeTab);
+
+  const returnToPreviousTab = () => {
+    if (previousTab) {
+      return setActiveTab(previousTab);
+    }
+  };
+
+  const onSaveButtonClick = () => {
+    if (activeTab === "save") {
+      returnToPreviousTab();
+    } else {
+      setActiveTab("save");
+    }
+  };
+
+  return (
+    <Box p="lg">
+      <Tabs value={activeTab} onTabChange={setActiveTab}>
+        <Group w="100%" position="apart">
+          <Tabs.List>
+            <Group>
+              <Tabs.Tab value="notebook">Notebook</Tabs.Tab>
+              <Tabs.Tab value="visualization">Visualization</Tabs.Tab>
+            </Group>
+          </Tabs.List>
+          {/* using a button instead of a tab for styling reasons */}
+          <Button onClick={onSaveButtonClick}>Save</Button>
+        </Group>
+
+        <Tabs.Panel value="notebook">
+          <Notebook onApply={() => setActiveTab("visualization")} />
+        </Tabs.Panel>
+        <Tabs.Panel value="visualization">
+          <QuestionVisualization />
+        </Tabs.Panel>
```

Comment:
There should be a special case to handle when users haven't clicked the `visualize` button yet as this shows the error `Question not found`
![image](https://github.com/user-attachments/assets/55028e8b-27b3-4342-a33c-49abf757dcd2)

In our app, it seems we don't show `Show visualization` button until we clicked `Visualize` button.

![image](https://github.com/user-attachments/assets/5165eaea-34de-4935-9b7b-9c503244695b)
![image](https://github.com/user-attachments/assets/4b093e10-5555-402c-8876-d5dbe37d3723)

Reply:
I think it might make more sense to always run the visualization when the user switches to the tab - then the user can either click `Visualize` or just go to the tab